### PR TITLE
[WIP] Deepstream DLR plugin

### PR DESCRIPTION
**Pull request by trevor-m**
This PR adds an example of how to use a Neo compiled model in an NVIDIA Deepstream application.

There are a few different components
1. NeoDLRLayer TensorRT Plugin (`neodlr_plugin.h`, `neodlr_plugin.cc`) - A TensorRT plugin layer which executes a Neo compiled model using DLR. This allows you to use a DLR model as a layer inside a TensorRT network.
2. Deepstream custom layer implementation (`nvdsinfer_custom_impl_neodlr.cc`, `config_infer_primary.txt`). Example of a deepstream stage which uses the NeoDLRLayer to execute a Neo compiled model. See `nvds_custom_impl.h` in Deepstream Docs/API for more info.

### CODE REVIEW

**Comment by apivovarov** [comment]
Existing C++ code in DLR uses `.cc` filename extension

File: examples/deepstream/neodlr_plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    std::vector<int64_t> shape(output_dim);
+    GetDLROutputShape(&model_, i, shape.data());
+    output_shapes_.emplace_back(shape);
+    // Type
+    const char* dtype;
+    GetDLROutputType(&model_, i, &dtype);
+    output_types_.emplace_back(dtype);
+  }
+  return 0;
+}
```
Comment (apivovarov) [inline examples/deepstream/neodlr_plugin.cpp:245]: no new line at the end of the file

Comment (trevor-m) [reply examples/deepstream/neodlr_plugin.cpp]: Thanks, fixed

**Comment by trevor-m** [comment]
> Existing C++ code in DLR uses `.cc` filename extension

Thanks for catching, fixed!

**Pull request closed without merging**